neuron.py

```diff
@@ -41,4 +45,4 @@ def calculate_error(self, expected_output: list[float], next_layer) -> float:
 
     def update_weights(self, learning_rate: float) -> None:
         delta_weights = np.multiply(self.__input, self.delta * learning_rate)
-        self.weights = np.add(self.weights, -delta_weights)
+        self.weights = np.clip(np.add(self.weights, -delta_weights), -10, 10)
```

Comment:
As we discussed clipping is problematic especially for the regression, but probably also for classification. Maybe we can at least clip depending on problem type until we find better solution?

Reply:
Will figure out later :)